src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseSimulationService.java

```diff
@@ -0,0 +1,157 @@
+
+        // The creation of the webhooks must occur after the initial push, because the participation is
+        // not yet saved in the database, so we cannot save the submission accordingly (see ProgrammingSubmissionService.notifyPush)
+        programmingExerciseScheduleService.scheduleExerciseIfRequired(programmingExercise);
+        groupNotificationService.notifyTutorGroupAboutExerciseCreated(programmingExercise);
+
+        return programmingExercise;
+    }
+
+    /**
+     * Sets the url and buildplan ids for the new exercise
+     * @param programmingExercise the new exercise
+     * @param exerciseRepoName the repo name of the new exercise
+     * @param testRepoName the test repo name of the new exercise
+     * @param solutionRepoName the solution repo name of the new exercise
+     */
+    private void setURLsAndBuildPlanIDsForNewExerciseWithoutLocalSetup(ProgrammingExercise programmingExercise, String exerciseRepoName, String testRepoName,
+            String solutionRepoName) {
+        final var projectKey = programmingExercise.getProjectKey();
+        final var templateParticipation = programmingExercise.getTemplateParticipation();
+        final var solutionParticipation = programmingExercise.getSolutionParticipation();
+        final var templatePlanName = TEMPLATE.getName();
+        final var solutionPlanName = SOLUTION.getName();
+        final var exerciseRepoUrl = "http://localhost:7990/scm/" + projectKey + "/" + exerciseRepoName + ".git";
+        final var testsRepoUrl = "http://localhost:7990/scm/" + projectKey + "/" + testRepoName + ".git";
+        final var solutionRepoUrl = "http://localhost:7990/scm/" + projectKey + "/" + solutionRepoName + ".git";
+        templateParticipation.setBuildPlanId(projectKey + "-" + templatePlanName);
+        templateParticipation.setRepositoryUrl(exerciseRepoUrl);
+        solutionParticipation.setBuildPlanId(projectKey + "-" + solutionPlanName);
+        solutionParticipation.setRepositoryUrl(solutionRepoUrl);
+        programmingExercise.setTestRepositoryUrl(testsRepoUrl);
+    }
+
+    /**
+     * This method creates the template and solution submissions and results for the new exercise
+     * These submissions and results are SIMULATIONS for the testing of programming exercises without local setup
+     * @param programmingExercise the new exercise
+     */
+    public void setupInitialSubmissionsAndResults(ProgrammingExercise programmingExercise) {
+        Optional<TemplateProgrammingExerciseParticipation> templateProgrammingExerciseParticipation = this.templateProgrammingExerciseParticipationRepository
+                .findByProgrammingExerciseId(programmingExercise.getId());
+        Optional<SolutionProgrammingExerciseParticipation> solutionProgrammingExerciseParticipation = this.solutionProgrammingExerciseParticipationRepository
+                .findByProgrammingExerciseId(programmingExercise.getId());
+        String commitHashBase = "abcdef01";
```

Comment:
I would prefer a commitHash that is as long as a normal SHA-1 commitHash (I believe 40 chars) as the commit hash might be displayed somewhere in the future and we should be consistent here.

Reply:
I added a random function, which creates a 40 char commit hash